File: backend/app/domain/schemas/applications/report_full_time.py
from typing import Any
from datetime import datetime

from pydantic import BaseModel, validator, Field

from app.domain.schemas.application import ApplicationResponse

class ReportFullTimeBase(BaseModel):
    from_full_time: bool
    full_time_id: int
    documents: list[Any] | None = Field(default_factory=list)

class ReportFullTimeCreate(ReportFullTimeBase):
    application_sub_type_id: int = 15

class ReportFullTimeUpdate(ReportFullTimeBase):
    application_sub_type_id: int = 15

class ReportFullTimeInDB(ReportFullTimeBase):
    pass

class ReportFullTimeResponse(ApplicationResponse):
    report_full_time: ReportFullTimeInDB

File: client/src/app/core/interfaces/applications/report-full-time.ts
import { Application } from "@interfaces/application";
import { file_path } from "@interfaces/documents";

export interface ReportFullTimeCreate {
    application_sub_type_id: 15;
    from_full_time: boolean;
    title:string;
    full_time_id: number;
    documents?: file_path[];
    justification: string;
}

export interface ReportFullTimeInDB extends ReportFullTimeCreate {
    id: string;
}

export interface ReportFullTimeResponse extends Application {
    report_full_time: ReportFullTimeInDB;
}